doc/source/serve/batching-guide.md

```diff
@@ -0,0 +1,46 @@
+# Requests Batching
+
+Serve offers a request batching feature that can improve your service throughput with sacrificing latency.
+
+This guide will teach you:
+- How to use Serve's `@serve.batch` decorator
+- How to configure `@serve.batch` decorator
+
+ML frameworks like Tensorflow, PyTorch, and Scikit-Learn are built to support efficiently evaluating multiple samples at the same time.
+Ray Serve allows you to take advantage of this feature via dynamic request batching.
+When a request arrives, it will wait in a queue. This queue buffers requests to form a batch of sample. The batch will then be picked up by the model for evaluation. After the evaluation, the result batch will be split up and send to the response individually.
+
+## Enable batching for your deployment
+You can enable batching by using the {mod}`ray.serve.batch` decorator. Let's take a look at an simple example. We will transform the MyModel class to accept a batch.
+```{literalinclude} doc_code/batching_guide.py
+---
+start-after: __single_sample_begin__
+end-before: __single_sample_end__
+---
+```
+
+The batching decorators expect you to make the following changes in your method signature:
+- The function should be an async method because the decorator batches in asyncio event loop.
```

Comment:
Make the following changes to the batch decorator method signature:

- The class method is declared as an async method because the decorator batches in asyncio event loop.